From akpm@linux-foundation.org Sun Apr 27 14:02:26 2008
From: Ciaran McCreesh <ciaran.mccreesh@googlemail.com>
Date: Sun, 27 Apr 2008 14:02:20 -0700
Subject: firmware loader: printk when requesting firmware
To: mm-commits@vger.kernel.org
Cc: ciaran.mccreesh@googlemail.com, greg@kroah.com
Message-ID: <200804272102.m3RL2KWR024665@imap1.linux-foundation.org>


From: Ciaran McCreesh <ciaran.mccreesh@googlemail.com>

Before requesting firmware, printk a message saying what we're requesting. This
makes it easier to see what's going on, and provides an explanation for the
huge silent delay that one would otherwise get after accidentally building
ipw2200 as a non-module.

Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/base/firmware_class.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/base/firmware_class.c
+++ b/drivers/base/firmware_class.c
@@ -398,6 +398,8 @@ _request_firmware(const struct firmware 
 
 	printk(KERN_INFO "firmware: requesting %s\n", name);
 
+	printk(KERN_INFO "firmware: requesting %s\n", name);
+
 	*firmware_p = firmware = kzalloc(sizeof(*firmware), GFP_KERNEL);
 	if (!firmware) {
 		printk(KERN_ERR "%s: kmalloc(struct firmware) failed\n",
